### ISSUE TITLE
Dozens of processes named "instance.purge" have been "delayed"  since 12/1/2016 and 1/5/2017

### ISSUE THREAD

**Issue by stefanlasiewski** [rancher/rancher#8316]
While chasing down repeated errors in the Rancher Server logs, I noticed that the redesigned "Processes" page at https://rancher.example.org/admin/processes/list?which=delayed show 23 processes which should have finished up months ago. None of them look like critical containers

How can I remove these things from Rancher? These instances are leaving thousands of Java stack traces in my Rancher Server logs, which makes the Rancher logs almost useless.

Please see this screenshot:

![image](https://cloud.githubusercontent.com/assets/142752/24380193/0c56cece-12ff-11e7-9d8e-f42d2cbff8f4.png)

---
Rancher 1.4.2 & Docker 1.13.1.

| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v1.4.2` Cattle: `v0.176.10` UI: `v1.4.9` |
|Access|`openldap` `admin`|
|Orchestration|`Cattle`|
|Route|`admin-tab.processes.list`|

**Comment by tfiduccia** [comment]
@stefanlasiewski Can you click on the host/instance ID then go to API and then purge them from there?

**Comment by stefanlasiewski** [comment]
If I view the host/instance ID in the API, the purge button is not clickable.

If I use the 'Delete' button on that screen, Rancher seems return an error message under "HTTP Response:"

```
{
"id": "3322515d-5452-441c-90eb-1d3541c605d5",
"type": "error",
"links": { },
"actions": { },
"status": 409,
"code": "Conflict",
"message": "Conflict",
"detail": null,
"baseType": "error"
}
```

**Comment by tfiduccia** [comment]
Then I think the only option you have now is to go into the db and remove the items from there.

**Comment by stefanlasiewski** [comment]
Are there instructions to do that cleanly?

Is this what the `rancher/cleanup` container is for?

**Comment by stefanlasiewski** [comment]
I managed to delete the 'host' items by hand, so those are a red herring. However, I'm unable to delete the 'instance:*' items.

**Comment by stefanlasiewski** [comment]
With Denise's help, I managed to fix this. Here's what I did:

1.  Run `docker run --rm rancher/cleanup-1-1:v0.1.2`. This removed a bunch of data from the DB and shrunk the DB size substantially, from 1.6G to 525M. However, the ghost processes still appeared in the Rancher UI and in the logs.
2. Upgraded from Rancher v1.4.2 to v1.5.7 on Tuesday. The ghost processes were gone immediately afterwards.

I assume that a combination of #1 & #2 together purged the data from the system. 

Closing ticket.
